torch/_inductor/config.py

```diff
@@ -308,6 +308,11 @@ def fx_graph_remote_cache_default():
     os.environ.get("TORCHINDUCTOR_COORDINATE_DESCENT_RADIUS", "1")
 )
 
+autoheuristic_mode = os.environ.get("TORCHINDUCTOR_AUTOHEURISTIC_MODE", "USE_HEURISTIC")
```

Comment:
Do we want to turn this on by default? cc: @eellison

Reply:
I only enabled it by default to be able to start a perf run, but I would probably be too scared to keep it enabled by default because I don't want to introduce any regressions.

Reply:
@AlnisM we could do something where if there are a large # of unique mm shapes and max-autotune is not set we use the heuristic. Wdyt ?

Reply:
Yea, can do in separate pr

Reply:
If people are willing to accept slightly worse performance in exchange for lower compile times when max-autotune is not set, then this sounds like a good idea to me.

I am collecting some data with the higher values for p_transposed right now which I think should make the heuristic perform better on "real" inputs.